packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts

```diff
@@ -24,11 +24,27 @@ export default async function route(nango: Nango, integration: ProviderConfig, h
     }
 
     if (Array.isArray(body)) {
-        const sorted = body.sort((a, b) => {
-            return a.occurredAt - b.occurredAt;
-        });
-        for (const event of sorted) {
-            await nango.executeScriptForWebhooks(integration, event, 'subscriptionType', 'portalId');
+        const groupedByObjectId = body.reduce((acc, event) => {
+            (acc[event.objectId] = acc[event.objectId] || []).push(event);
+            return acc;
+        }, {});
+
+        for (const objectId in groupedByObjectId) {
+            const sorted = groupedByObjectId[objectId].sort((a: any, b: any) => {
+                const aIsCreation = a.subscriptionType.endsWith('.creation');
+                const bIsCreation = b.subscriptionType.endsWith('.creation');
+                if (aIsCreation && !bIsCreation) {
+                    return -1;
+                }
+                if (!aIsCreation && bIsCreation) {
+                    return 1;
+                }
+                return a.occurredAt - b.occurredAt;
```

Comment:
would something like the following simpler?
```
const aIsCreation = a.subscriptionType.endsWith('.creation') ? 1 : 0;
const bIsCreation = b.subscriptionType.endsWith('.creation') ? 1 : 0;
return bIsCreation - aIsCreation || a.occurredAt - b.occurredAt;
```

also maybe having a test for this function might be useful

Reply:
Thanks, updated with https://github.com/NangoHQ/nango/pull/1529/commits/ad0995ffc09df6676478dcc69a4927a6a0344e6a